File: TP3/JAVA/src/test/resources/TestFactory.java
package test.resources;

import org.junit.jupiter.api.Test;

import src.Etudiant;

import static org.junit.jupiter.api.Assertions.*;

public class TestFactory {
    
    @Test
    def testPersonneFactory() {
        IPersonne prof = PersonneFactory.creer("enseignant", "Durand", 45, 4000);
        assertTrue(isinstance(prof, Enseignant));
        assertEquals(prof.getNom(),"Durand");
        
        IPersonne eleve = PersonneFactory.creer("etudiant", "Alice", 20, "E001", List.of("Python"));
        assertTrue(isinstance(eleve, Etudiant));
        assertEquals(eleve.getNom(), "Alice");
        assertEquals(eleve.getAge(), 20);
        assertEquals(eleve.getNumeroEtudiant(), "E001");
        assertEquals(eleve.getCours(), List.of("Python"));
    }
}

File: TP3/PYTHON/test/test_factory.py
from src.codes_initiaux.enseignant import Enseignant
from src.pattern.Factory.PersonneFactory import PersonneFactory

def test_personne_factory():
    prof = PersonneFactory.creer_personne("enseignant", "Durand", 45, [4000])
    assert isinstance(prof, Enseignant)
    assert prof.get_nom() == "Durand"
    eleve = PersonneFactory.creer_personne("etudiant", "Alice", 20, ["E001", ["Python"]])
    assert eleve.get_nom() == "Alice"
    assert eleve.get_age() == 20
    assert eleve.numero_etudiant == "E001"
    assert eleve.cours == ["Python"]
